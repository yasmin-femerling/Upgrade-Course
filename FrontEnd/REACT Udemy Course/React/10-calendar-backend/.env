PORT=3001
DB_CNN=mongodb+srv://mern_user:K6Z2h49lxF50QJsf@cluster0.dtvdp.mongodb.net/mern_calendar
SECRET_JWT_SEED=Esto-Es-Una_Palabra_Secreta123456