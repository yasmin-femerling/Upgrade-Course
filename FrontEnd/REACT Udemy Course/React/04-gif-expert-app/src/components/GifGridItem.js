import PropTypes from 'prop-types';
import React from 'react';

export const GifGridItem = ({ title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p> { title } </p>
        </div>
    )
}

/* tarea prueba
mostrar componente correctamente
    * shallow
    *wrapper
    * wrapper .tomatchsnapshot()
*/


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};