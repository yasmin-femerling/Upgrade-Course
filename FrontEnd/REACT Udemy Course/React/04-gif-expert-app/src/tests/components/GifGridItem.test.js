import {shallow} from 'enzyme';
import { hasUncaughtExceptionCaptureCallback } from 'process';
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://random.com';
    
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

  test('debe mostrar correctamente el componente ', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('debe tener un parrafo con el title', () => {
    
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });
  
  test('debe tener la imagen igual al url y alt de props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  });

  test('debe tener animate_fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    console.log(className)
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
  
  
  
});
