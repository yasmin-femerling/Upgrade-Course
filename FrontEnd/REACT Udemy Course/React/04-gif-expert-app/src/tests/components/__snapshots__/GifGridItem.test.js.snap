// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifGridItem debe mostrar correctamente el componente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://random.com"
  />
  <p>
     
    Un titulo
     
  </p>
</div>
`;
